src/vivarium/framework/engine.py

```diff
@@ -207,6 +207,10 @@ def __init__(
     def name(self) -> str:
         return self._name
 
+    @property
+    def results(self) -> Dict[str, Any]:
```

Comment:
Would it maybe be less confusing to have a method `get_results()` than to have this as a property? Since `sim._results` is very different from `sim.results`. I think that's more consistent with how we get things from a Simulation anyway, (e.g. `sim.get_value()` or `sim.get_population()`).